### ISSUE TITLE
HIP-SYCL Interop Failures

### ISSUE THREAD

**Issue by pvelesko** [CHIP-SPV/chip-spv#297]
```
CHIP_BE=level0 ./samples/hip_sycl_interop/hip_sycl_interop
terminate called after throwing an instance of 'sycl::_V1::runtime_error'
  what():  Native API failed. Native API returns: -33 (PI_ERROR_INVALID_DEVICE) -33 (PI_ERROR_INVALID_DEVICE)
Aborted
```

**Comment by Sarbojit2019** [comment]
@pvelesko,
May I know in which branch do you see the crash? I synced latest chip-spv : 'main' but don't see the failure with debug build.

sarbojit@aelab299:~/src/chip-spv/build$ **CHIP_BE=level0 ./samples/hip_sycl_interop/hip_sycl_interop**
CHIP debug [TID 320179] [1673277017.934195697] : Reading CHIP_PLATFORM from env
CHIP debug [TID 320179] [1673277017.934293539] : Reading CHIP_DEVICE_TYPE from env
CHIP debug [TID 320179] [1673277017.934302908] : Reading CHIP_DEVICE from env
CHIP debug [TID 320179] [1673277017.934311739] : Reading CHIP_BE from env
CHIP debug [TID 320179] [1673277017.934335773] : CHIP_PLATFORM=0
CHIP debug [TID 320179] [1673277017.934343416] : CHIP_DEVICE_TYPE=gpu
CHIP debug [TID 320179] [1673277017.934351635] : CHIP_DEVICE=0
CHIP debug [TID 320179] [1673277017.934357897] : CHIP_BE=level0
CHIP debug [TID 320179] [1673277017.934366494] : CHIPDriver Initialize
CHIP debug [TID 320179] [1673277017.934375624] : CHIPBE=LEVEL0... Initializing Level0 Backend
CHIP debug [TID 320179] [1673277017.934412604] : CHIPBackend Base Constructor
CHIP debug [TID 320179] [1673277017.945499748] : CHIPQueue() 0x55a2155b78a0
CHIP debug [TID 320179] [1673277017.945687415] : 0x55a2151aa010 CHIPContext::addDevice() 0x55a2155b71c0
CHIP debug [TID 320179] [1673277017.945757722] : Thread Created with ID : 140253797992192
CHIP debug [TID 320179] [1673277017.945797028] : Thread Created with ID : 140253789599488
CHIP debug [TID 320179] [1673277017.945818761] : Reading CHIP_JIT_FLAGS from env
CHIP debug [TID 320179] [1673277017.945828689] : __hipRegisterFatBinary
CHIP debug [TID 320179] [1673277017.945851462] : Bundle entry ID 0 is: 'host-x86_64-unknown-linux'

CHIP debug [TID 320179] [1673277017.945862737] : Not a SPIR-V triple, ignoring

CHIP debug [TID 320179] [1673277017.945870326] : Bundle entry ID 1 is: 'hip-spirv64----generic'

CHIP debug [TID 320179] [1673277017.945889308] : Register module: 0x55a2155b5ab0

CHIP debug [TID 320179] [1673277017.945896628] : CHIPBackend->register_module()
CHIP debug [TID 320179] [1673277017.946122402] : JIT compiler flags: -cl-std=CL2.0 -cl-take-global-address -cl-match-sincospi
CHIP debug [TID 320179] [1673277017.996067586] : hipGetCHIPBackend
CHIP debug [TID 320179] [1673277017.996122703] : CHIPQueue() 0x55a2155d9860
CHIP debug [TID 320179] [1673277017.996219492] : 0x55a2155b71c0 CHIPDevice::addQueue(0x55a2155d9860)
CHIP debug [TID 320179] [1673277017.996261298] : CHIPQueue 0x55a2155d9860 added to the queue vector for device 0x55a2155b71c0
CHIP debug [TID 320179] [1673277017.996267251] : hipGetBackendNativeHandles
Verify results between OneMKL & Serial: SUCCESS - The results are correct!
CHIP debug [TID 320179] [1673277018.768563666] : Unregister module: 0x55a2155b5ab0

CHIP debug [TID 320179] [1673277018.768607611] : CHIPBackend->unregister_module()
CHIP debug [TID 320179] [1673277018.768648708] : __hipUnRegisterFatBinary 0

CHIP debug [TID 320179] [1673277018.768671876] : Uninitializing CHIP...
CHIP debug [TID 320179] [1673277019.268920317] : Setting LastEvent for 0x55a2155b78a0 0x0 -> 0x0
CHIP warning [TID 320179] [1673277019.269041466] : Not all user created streams have been destoyed... Queues remaining: 1
CHIP warning [TID 320179] [1673277019.269075526] : Make sure to call hipStreamDestroy() for all queues that have been created via hipStreamCreate()
CHIP warning [TID 320179] [1673277019.269098706] : Removing user-created streams without calling a destructor
CHIP debug [TID 320179] [1673277019.293016248] : CHIPBackend Destructor. Deleting all pointers.
CHIP debug [TID 320179] [1673277019.293136876] : 0x55a2151aa010 CHIPContext.removeDevice() 0x55a2155b71c0
CHIP debug [TID 320179] [1673277019.293208332] : ~CHIPDevice() 0x55a2155b71c0
CHIP debug [TID 320179] [1673277019.293265482] : Setting LastEvent for 0x55a2155b78a0 0x0 -> 0x0
CHIP debug [TID 320179] [1673277019.293316310] : Setting LastEvent for 0x55a2155b78a0 0x0 -> 0x0
CHIP debug [TID 320179] [1673277019.293485408] : ~CHIPContext() 0x55a2151aa010

**Comment by pvelesko** [comment]
@Sarbojit2019 this is on main 

Intel oneAPI SDK 2022.3 Public + Engineering Components (2022.10.15.006)

**Comment by Sarbojit2019** [comment]
Quick update : I could reproduce the issue with latest SDK 2023.0.0.20221201. Crash is happening in the sample app not in the CHIP-SPV runtime. It seems there is some change in latest sycl interop which is why test is failing with latest SDK. I will continue looking into it and update my findings.

**Comment by Sarbojit2019** [comment]
Fix raised #320
